src/main/java/bio/terra/common/logging/RequestIdFilter.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.common.logging;
+
+import static org.apache.commons.lang3.ObjectUtils.getFirstNonNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.concurrent.ThreadLocalRandom;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.hashids.Hashids;
+import org.slf4j.MDC;
+
+/**
+ * A Servlet filter that ensures a requestId is extracted from an inbound HTTP request, or otherwise
+ * generated randomly.
+ *
+ * <p>The requestId is added to the MDC context and set as a response header.
+ */
+public class RequestIdFilter implements Filter {
+
+  public static final String REQUEST_ID_HEADER = "X-Request-ID";
+  public static final String REQUEST_ID_MDC_KEY = "requestId";
+
+  private final Hashids hashids = new Hashids("requestIdSalt", 8);
```

Comment:
Why Hashid here? Why do we need a special library to turn a random int into a string?

Reply:
I'm porting code from WSM – presumably that passed code review at some point, and someone thought it was necessary. I'll plan to loop Isaac in for comment once this is past draft mode.

Reply:
@ansingh7115 Any knowledge about this? I don't _think_ it was mine?

Reply:
I flagged this in my review of WSM as a question. What are the requirements of the id?
Would it be better as a ShortUUID so it would be guaranteed unique?

Anyway, it shouldn't stop merging this, but I would like to know. I'm happy to swing by and fix it when I do the UUID work.